lib/common/error-utils.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const constructors = new Map([
+  [Error.name, Error],
+  [EvalError.name, EvalError],
+  [RangeError.name, RangeError],
+  [ReferenceError.name, ReferenceError],
+  [SyntaxError.name, SyntaxError],
+  [TypeError.name, TypeError],
+  [URIError.name, URIError]
+])
+
+exports.deserialize = function (error) {
+  if (error.__ELECTRON_SERIALIZED_ERROR__ && constructors.has(error.name)) {
+    const constructor = constructors.get(error.name)
+    const rehydratedError = new constructor(error.message)
+    rehydratedError.stack = error.stack
+    return rehydratedError
```

Comment:
I know it's not new to this PR, but `rehydratedError` is a really weird variable name. While we're here, could you rename this `deserializedError`?

Reply:
@ckerr done